src/render/Renderer.cpp

```diff
@@ -617,6 +619,11 @@ void CHyprRenderer::renderLayer(SLayerSurface* pLayer, CMonitor* pMonitor, times
 
     g_pHyprOpenGL->m_pCurrentLayer = pLayer;
 
+    if (*PDIMAROUND && pLayer->dimAround && !m_bRenderingSnapshot) {
```

Comment:
de-dup: why not put this before the first check instead of duping this here and in renderSnapshot?

Reply:
Hmm. Doing that sorta works, but the dimaround flashes more-opaque at the start of the fade out and I can't tell why (it's in a separate branch against main [here](https://github.com/TheOnlyMrCat/Hyprland/commit/bfdee9f0d0cb6414fe4e9c9a606e9f375679eca9#diff-dbcd1ce1e0b916536f17c999473ad9e22c014646ddb2df15c45d726884a3481c) until I can figure that out).

I did it duplicated here because it was duplicated for the window dimaround. If I can get the dedup working here, should I try and deduplicate the code there as well?

Reply:
Ok I figured out why it was flashing and it's no longer doing it. I haven't done the dedup on the window code because it'd just result in more duplication again.